maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/util/PluginUtils.java

```diff
@@ -63,7 +63,14 @@ private PluginUtils()
         mavenComponents.put( "org.apache.maven.plugin.MojoExecution", "${mojoExecution}" );
         mavenComponents.put( "org.apache.maven.plugin.descriptor.PluginDescriptor", "${plugin}" );
         mavenComponents.put( "org.apache.maven.settings.Settings", "${settings}" );
-        
+
+        mavenComponents.put( "org.apache.maven.api.Session", "${session}" );
+        mavenComponents.put( "org.apache.maven.api.Project", "${project}" );
+        mavenComponents.put( "org.apache.maven.api.MojoExecution", "${mojoExecution}" );
+        // TODO: apiv4: add PluginDescriptor to the api ?
+        //mavenComponents.put( "org.apache.maven.api.descriptor.PluginDescriptor", "${plugin}" );
+        mavenComponents.put( "org.apache.maven.api.settings.Settings", "${settings}" );
```

Comment:
I'm not sure about it ...

Reply:
As I remember `@Component Session session` is translated to  `@Parameter( property = "${session}" )` - to check

Reply:
Yes, and the map is here to do that.  So if I remove the population of the `mavenComponents` with objects from the v4 api, this won't work anymore.  The IT actually tests that.

Reply:
Also, I don't think the use of `@Parameter` or `@Component` should be dictated by the implementation.  For example, the maven session is available as both a parameter injected by the plugin manager and a sisu bean inside the `@SessionScope`.  Also `@Component` relates to the sisu components, but this does not have to be made explicit.
I would personally favor the `@Component` approach for everything that does not come from the configuration.

Reply:
Me too.